### PR TITLE
BLO-819

### PR DESCRIPTION
**Pull request by bluecco**
### Issue / feature description

`Privacy statement not working in ErrorBoundary component`

### Changes

- added router propr to `ErrorBoundary`
- reset error props on `ErrorBoundary` if next link is privacy settings 

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

File: packages/extension/src/ui/components/ErrorBoundary.tsx [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+function withRouter(Component: any) {
+  function ComponentWithRouterProp(props: any) {
+    const location = useLocation()
+    const navigate = useNavigate()
+    const params = useParams()
+    return <Component {...props} router={{ location, navigate, params }} />
+  }
+
+  return ComponentWithRouterProp
+}
```
Comment (simonheys) [inline packages/extension/src/ui/components/ErrorBoundary.tsx:16]: Guessing this HoC may be useful elsewhere in future - can split into a separate file?

File: packages/extension/src/ui/components/ErrorBoundary.tsx [hunk 2]
```diff
@@ -32,10 +59,13 @@ export class ErrorBoundary extends Component<
   }
 
   render() {
+    console.log(this.props)
```
Comment (simonheys) [inline packages/extension/src/ui/components/ErrorBoundary.tsx:62]:
```suggestion
```

File: packages/extension/src/ui/components/ErrorBoundary.tsx [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
+    return <Component {...props} router={{ location, navigate, params }} />
+  }
+
+  return ComponentWithRouterProp
+}
 
 interface ErrorBoundaryProps {
   fallback: React.ReactElement
   children: React.ReactNode
+  router: any
```
Comment (diegodelrieu) [inline packages/extension/src/ui/components/ErrorBoundary.tsx:21]: is router actually any ? also isnt this an opportunity to get rid of this HoC and replace with function component ?

File: packages/extension/src/ui/components/ErrorBoundary.tsx (hunk 1 above)
Comment (bluecco) [reply packages/extension/src/ui/components/ErrorBoundary.tsx:16]: sure, I didn't thought about it

File: packages/extension/src/ui/components/ErrorBoundary.tsx (hunk 3 above)
Comment (bluecco) [reply packages/extension/src/ui/components/ErrorBoundary.tsx:21]: fixing router type! forgot to update it @diegodelrieu 🙏 

about HoC, the refactor would require an additional npm package.
iirc, we cannot use a functional component for ErrorBoundary but only class, mainly due to `componentDidCatch`

https://reactjs.org/docs/error-boundaries.html

**Review by diegodelrieu — APPROVED** [review]

**Review by simonheys — APPROVED** [review]

**Pull request merged**